### DISCUSSION TITLE
API for integrations?  (mostly editor)

### DISCUSSION THREAD

**Discussion by trharmon0** [FreeTubeApp/FreeTube#6168] [General]
Howdy. Is there an API provider available for doing external (to the process) integrations. An example would be scripting various things against FreeTube using a programming editor?

I've read on Local API but that seems like it might be just for extracting and perhaps not for general scripting. Also, I've yet to find docs on whether the Local API provides an API provider listen port (ex: localhost:3000). 

thx

**Comment by absidue** [comment]
No there isn't and stuff like that is out of scope for FreeTube. If you want to customise stuff in FreeTube, we recommend modifying the source code and creating a custom build for yourself.

If you want to get data from YouTube yourself, you should use the YouTube.js library directly yourself, as that is the library that powers FreeTube's local API.

**Reply by trharmon0** [reply]
Thank you for confirming. 

I am in the process of learning Typescript/Deno so its not outside the realm of possibilities that I might send in a few PRs.

thx again

**Reply by absidue** [reply]
You are welcome to submit pull requests for other things, we likely would not accept any pull requests for things like this, hence the "out of scope" and the "create a custom build for yourself" parts of my message :).

**Reply by trharmon0** [reply]
To be clear, PRs to add functionality to core as opposed to my original plan of "plugins" which do their work via an API provider. I think we are on the same page.

**Reply by absidue** [reply]
In that case it would of course depend on what changes you are planning on doing, the fact that your original thought was that the ideas would require FreeTube to expose an HTTP API concerns me. That being said you are of course still welcome to create pull requests, if you think that other people would appreciate your changes.

**Reply by trharmon0** [reply]
It could have just as easily been a UDS. I'm just hoping to be able to monitor + manipulate download queue and play queue from programming editor.
